src/swell/suites/base/cylc_workflow.py

```diff
@@ -0,0 +1,89 @@
+# (C) Copyright 2021- United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+
+# --------------------------------------------------------------------------------------------------
+
+from typing import Tuple
+from abc import abstractmethod
+
+from swell.utilities.logger import get_logger
+
+# --------------------------------------------------------------------------------------------------
+
+
+header_str = '''#!jinja2
+# (C) Copyright 2021- United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+'''
+
+
+class CylcWorkflow():
+
+    '''
+    Handles generating the flow.cylc file contents using the CylcSection syntax for each
+    necessary section in the cylc file. Since Swell workflows share a lot of common language,
+    this method has the convenience of automatically setting a lot of the contents. This means
+    that the graph section is the only part that will need to be adjusted in many cases,
+    and tasks may need to be altered in src/swell/tasks/task_runtimes.py.
+    '''
+
+    def __init__(self, experiment_dict, slurm_external) -> None:
+        self.experiment_dict = experiment_dict
+        self.slurm_external = slurm_external
+
+        self.logger = get_logger(self.__class__.__name__)
+
+        self.tasks = []
+        self.set_tasks()
+
+    # --------------------------------------------------------------------------------------------------
+
+    def default_header(self) -> str:
+        return header_str
+
+    # --------------------------------------------------------------------------------------------------
+
+    @abstractmethod
+    def set_tasks(self) -> None:
+        pass
+
+    # --------------------------------------------------------------------------------------------------
+
+    def get_independent_and_model_tasks(self) -> Tuple[list, dict]:
```

Comment:
See comment above. You can drop the `typing.Tuple` import and just do:

```suggestion
    def get_independent_and_model_tasks(self) -> tuple[list, dict]:
```